backend/app/readers/all_source_reader.py

```diff
@@ -0,0 +1,22 @@
+from collections.abc import Generator
+
+from app.models.record import Record
+from app.models.settings import settings
+from app.readers.create_readers import create_readers
+from app.readers.reader.reader import Reader
+
+
+class AllSourceReader(Reader):
+    """A multiplexer for different Readers.
+
+    Read in output data and yield them as Records.
+    """
+
+    def __init__(self) -> None:
+        self.__readers: tuple[Reader | None, ...] = create_readers(settings=settings)
+
+    def read(self) -> Generator[Record, None, None]:
+        """Read in output data and yield Records."""
+        for reader in self.__readers:
+            if reader:
```

Comment:
Without the `None` reader you don't need this check.

Reply:
I have removed the check from `AllSourceReader.read`.